specfile/specfile.py

```diff
@@ -111,6 +152,45 @@ def __exit__(
     ) -> None:
         self.save()
 
+    def __deepcopy__(self, memodict: Dict[int, Any]):
+        """
+        Deepcopies the object, handling file-like attributes.
+        """
+        specfile = self.__class__.__new__(self.__class__)
+        memodict[id(self)] = specfile
+
+        for k, v in self.__dict__.items():
+            if k == "_file":
+                continue
+            setattr(specfile, k, copy.deepcopy(v, memodict))
+
+        try:
+            path = Path(cast(FileIO, self._file).name)
+        except AttributeError:
+            # not a named file, try `getvalue()`
+            try:
+                specfile._file = type(cast(FileIOType, self._file))(
+                    cast(FileIOType, self._file).getvalue()
+                )
+            except AttributeError:
+                # last resort, try to instantiate without arguments
+                # and copy data
+                specfile._file = type(self._file)()
+                specfile._file.write(self._file.read())
+                specfile._file.seek(0)
+        else:
+            try:
+                specfile._file = path.open(
+                    mode=cast(TextIOBase, self._file).mode,
+                    encoding=cast(TextIOBase, self._file).encoding,
+                    errors=cast(TextIOBase, self._file).errors,
+                )
```

Comment:
```suggestion
                specfile._file = cast(TextIOWrapper,
                    path.open(
                    mode=cast(TextIOBase, self._file).mode,
                    encoding=cast(TextIOBase, self._file).encoding,
                    errors=cast(TextIOBase, self._file).errors,
                ))
```
This was needed to silence,
`Incompatible types in assignment (expression has type "IO[Any]", variable has type "IOBase")  [assignment] (185:33)`